### PR TITLE
Web3 types

### PR DESCRIPTION
**Pull request by joshstevens19**
web3 types with some fixing to some other types

### CODE REVIEW

File: packages/web3-providers/types/index.d.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export class SocketProviderAdapter extends AbstractProviderAdapter {
 }
 
 export class BatchRequest {
+    constructor();
     constructor(provider: provider, jsonRpcMapper: JsonRpcMapper, jsonRpcResponseValidator: JsonRpcResponseValidator);
```
Comment (nivida) [inline packages/web3-providers/types/index.d.ts:59]: That's not the current constructor but I will change it by my self after I wrote all the providers tests.

File: packages/web3/types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import {AbstractWeb3Module, Web3ModuleOptions, Providers} from 'web3-core';
+import {Utils} from 'web3-utils';
+import * as net from 'net';
```
Comment (nivida) [inline packages/web3/types/index.d.ts:22]: I think ```import * as net from 'net'``` is not needed this is just for the nodejs runtime and you will have there the types.

File: packages/web3/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import {provider, AbstractProviderAdapter, BatchRequest} from 'web3-providers';
+import {Eth} from "web3-eth";
+import {Network} from "web3-net";
```
Comment (nivida) [inline packages/web3/types/index.d.ts]: Could you run eslint for the typing files to change the double quotes to single quotes and other stuff? :)

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-providers/types/index.d.ts (hunk 1 above)
Comment (joshstevens19) [reply packages/web3-providers/types/index.d.ts:59]: that constructor was what it as before, yeah if you can change it what would be great 👍

File: packages/web3/types/index.d.ts (hunk 3 above)
Comment (joshstevens19) [reply packages/web3/types/index.d.ts]: oops normally do will sort now

File: packages/web3/types/index.d.ts (hunk 2 above)
Comment (joshstevens19) [reply packages/web3/types/index.d.ts:22]: this is a types file, i am not sure that would work? I think you need to `import * as net from 'net'`. For example even though fs is built into node you still need to import it to use it. Think it has to stay as it is and i think should as its more explicit of what namespace its coming from.. thoughts?

**Review by nivida — APPROVED** [review]

**Pull request merged**